daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -243,6 +250,18 @@ class DataProcessor private (
 
   def withValidationMode(mode:ValidationMode.Type): DataProcessor = copy(validationMode = mode)
 
+  def withValidator(validator: Validator): DataProcessor = copy(validationMode = ValidationMode.Custom(validator))
+
+  lazy val validator: Validator = {
```

Comment:
This is the validator factory right? rename.

Reply:
No this is the cached Validator instance.

Reply:
Right. Maybe I need more coffee today.